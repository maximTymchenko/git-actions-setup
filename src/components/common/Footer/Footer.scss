@import '../../../styles/vars';

.footer {
    &__wrapper {
        display: flex;
        justify-content: space-between;
        flex-wrap: wrap;
        @include section__size;
        margin: 0 auto;
        @include main__padding;

        @media screen and (min-width: $mac__13) {
            margin: 100px auto 0;
            // position: relative;
            // right: 120px;
        }

        @media screen and (max-width: $desktop) {
            flex-direction: column;
            padding-right: 30px;
        }

        @media screen and (max-width: $tablet) {
            margin-left: 0;
            padding-right: 20px;
        }
    }
    &__info {
        display: flex;
        align-items: flex-start;
        border-top: 2px solid $secondary;
        padding-top: 20px;

        @media screen and (max-width: $desktop) {
            width: 100%;
            justify-content: space-between;
            margin-bottom: 50px;
        }
    }

    &__functions {
        border-top: 2px solid $secondary;
        padding-top: 20px;
        display: flex;

        @media screen and (max-width: $desktop) {
            margin-top: 25px;
            justify-content: space-between;
        }

        @media screen and (max-width: $tablet) {
            justify-content: space-around;
        }

        @media screen and (max-width: $mobile) {
            flex-wrap: wrap;
            justify-content: space-between;
        }
    }

    &__country {
        display: block;
        margin-bottom: 10px;
        font-family: 'OktaNeue-Black';
        font-size: 24px;
        font-weight: 900;
        letter-spacing: 0.4px;
        color: $text;
    }
    &__timezone {
        display: block;
        margin-bottom: 5px;
        font-family: 'OktaNeue-Black';
        font-size: 11px;
        font-weight: 900;
        color: $text__body;
    }

    &__phone,
    &__adress,
    &__index {
        display: block;
        text-decoration: none;
        font-family: 'OktaNeue-Normal';
        font-size: 16px;
        line-height: 1.69;
        color: $text__body;
    }
    &__index {
        margin-bottom: 20px;
    }

    &__location {
        margin-right: 50px;

        @media screen and (max-width: $mobile) {
            margin-right: 0;
        }
    }

    &__links {
        display: flex;
        flex-direction: column;
        justify-content: center;
    }

    &__logo {
        margin-bottom: 15px;
        margin-left: 6px;
        cursor: pointer;
    }
    &__social {
        text-decoration: none;
        cursor: pointer;
        display: inline-block;
        font-family: 'OktaNeue-Black';
        font-size: 20px;
        font-weight: 900;
        letter-spacing: 0.4px;
        color: $secondary;

        &-wrapper {
            display: flex;
            flex-direction: row;
            align-items: center;
            margin-left: 20px;
        }
    }

    &__bar {
        margin-right: 10px;
        margin-left: 10px;
        width: 3px;
        height: 23px;
        background-color: $section__border;
    }

    &__lists {
        margin-right: 50px;

        &:last-child {
            margin-right: 0;
        }

        @media screen and (max-width: $mobile) {
            flex-basis: 35%;
            margin-bottom: 25px;
        }
    }

    &__link {
        font-family: 'OktaNeue-Medium';
        font-size: 15px;
        font-weight: 500;
        line-height: 1.53;
        color: $text__body;
        cursor: pointer;
        margin-bottom: 5px;

        &:hover {
            color: $primary__light;
        }

        & span {
            font-family: 'OktaNeue-Black';
            font-size: 10px;
            font-weight: 900;
            letter-spacing: 0.5px;
            text-align: center;
            color: $background__primary;
            background-color: $color__error;
            padding: 0 3px;
            margin-left: 3px;
        }
        &:first-child {
            font-family: 'OktaNeue-Black';
            font-size: 15px;
            font-weight: 900;
            letter-spacing: 0.1px;
            color: $text;
            margin-bottom: 6px;
        }
    }
}
