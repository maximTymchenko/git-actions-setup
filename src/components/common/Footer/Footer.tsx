import React, { FC, useState } from 'react';
import { Link } from 'gatsby';
import Button from '../Button';
import Clock from '../Clock';
import SmallFooter from '../SecondaryFooter';
import './Footer.scss';
import blackBird from '../../../assets/img/bird-logo.svg';
import arrowRight from '../../../assets/img/arr-right.svg';

type TFooterServices = {
    activeServiece: null | any;
    services: {
        id: number;
        title: string;
        page: string;
    }[];
};

const Footer: FC = () => {
    const [findServices, setFindServices] = useState<TFooterServices>({
        activeServiece: null,
        services: [
            { id: 1, title: 'Web development', page: '/what-we-do/web-development/' },
            { id: 2, title: 'Quality Control' },
            { id: 3, title: 'Research' },
            { id: 4, title: 'Analytics' },
            { id: 5, title: 'Cloud Services', page: '/what-we-do/cloud-services/' },
            { id: 6, title: 'Mobile development', page: '/what-we-do/mobile-development/' },
        ],
    });

    const [pickAnExpertise, setPickAnExpertise] = useState<TFooterServices>({
        activeExpertise: null,
        expertises: [
            { id: 1, title: 'Retail and E-commerce', page: '/what-we-do/retail-and-e-commerce/' },
            { id: 2, title: 'Construction' },
            { id: 3, title: 'Finance and ERP' },
            { id: 4, title: 'Media' },
            { id: 5, title: 'AR & VR', page: '/what-we-do/ar-&-vr/' },
        ],
    });

    const [company, setCompany] = useState<TFooterServices>({
        activeCompany: null,
        companies: [
            { id: 1, title: 'About Us' },
            { id: 2, title: 'Blog and Resources', page: '/blog/' },
            { id: 3, title: 'Careers', label: 'Hiring', isLabled: true },
            { id: 4, title: 'Contact Blackbird', page: '/contact-us/' },
        ],
    });

    return (
        <footer data-testid="footer" className="footer">
            <div className="footer__wrapper">
                <div className="footer__info">
                    <div className="footer__location">
                        <span className="footer__country">Ukraine HQ</span>
                        <span className="footer__timezone">+2 UTC/EET</span>
                        <Clock timezone={'Europe/Kiev'} />
                        <a href="tel:+380–93–104–27–57" className="footer__phone">
                            +380–93–104–27–57
                        </a>
                        <span className="footer__adress"> 16 Shota Rustaveli St Kyiv,</span>
                        <span className="footer__index">02000</span>
                        <Link className="link" to="/contact-us">
                            <Button
                                icon={arrowRight}
                                type={'primary'}
                                className="button"
                                children={'Get in touch'}
                                desc={'Get in touch'}
                            />
                        </Link>
                    </div>
                    <div className="footer__links">
                        <Link to="/" className="link">
                            <img src={blackBird} alt="" className="footer__logo" />
                        </Link>
                        <div className="footer__social-wrapper">
                            <a
                                target="_blank"
                                href="https://ua.linkedin.com/company/blackbird-lab"
                                className="footer__social"
                            >
                                in
                            </a>
                            <div className="footer__bar"></div>
                            <a
                                target="_blank"
                                href="https://jobs.dou.ua/companies/blackbird-lab/"
                                className="footer__social"
                            >
                                DOU
                            </a>
                        </div>
                    </div>
                </div>
                <div className="footer__functions">
                    <ul className="footer__lists">
                        <span className="footer__small-title">Find a Service</span>
                        {findServices.services.map((service, index) => (
                            <Link to={service.page}>
                                <li key={index} className="footer__link">
                                    {service.title}
                                </li>
                            </Link>
                        ))}
                    </ul>
                    <ul className="footer__lists">
                        <span className="footer__small-title">Pick an Expertise</span>
                        {pickAnExpertise.expertises.map((expertise, index) => (
                            <Link to={expertise.page}>
                                <li key={index} className="footer__link">
                                    {expertise.title}
                                </li>
                            </Link>
                        ))}
                    </ul>
                    <ul className="footer__lists">
                        <span className="footer__small-title">Company</span>
                        {company.companies.map((company, index) => (
                            <Link to={company.page}>
                                <li key={index} className="footer__link">
                                    {company.title}
                                    {company.isLabled && (
                                        <span className="footer__label">{company.label}</span>
                                    )}
                                </li>
                            </Link>
                        ))}
                    </ul>
                </div>
            </div>
            <SmallFooter />
        </footer>
    );
};

export default Footer;
