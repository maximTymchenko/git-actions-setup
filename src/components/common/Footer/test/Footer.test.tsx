import React from 'react'
import Footer from '../Footer'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(<Footer></Footer>)
  expect(getByTestId('footer')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer.create(<Footer></Footer>).toJSON()
  expect(tree).toMatchSnapshot()
})
