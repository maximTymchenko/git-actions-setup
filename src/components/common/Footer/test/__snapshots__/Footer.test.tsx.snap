// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<footer
  className="footer"
  data-testid="footer"
>
  <div
    className="footer__wrapper"
  >
    <div
      className="footer__info"
    >
      <div
        className="footer__location"
      >
        <span
          className="footer__country"
        >
          Ukraine HQ
        </span>
        <span
          className="footer__timezone"
        >
          +2 UTC/EET
        </span>
        <span
          className="clock"
          data-testid="clock"
        >
          03:01 PM
        </span>
        <a
          className="footer__phone"
          href="tel:+380–93–104–27–57"
        >
          +380–93–104–27–57
        </a>
        <span
          className="footer__adress"
        >
           16 Shota Rustaveli St Kyiv,
        </span>
        <span
          className="footer__index"
        >
          02000
        </span>
        <a
          className="link"
          href="/contact-us"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <button
            className="button button--primary button--undefined button--undefined"
            data-testid="button"
          >
            Get in touch
            <img
              alt="Get in touch"
              className="button__icon button__icon--undefined"
              data-testid="button__icon"
              src="test-file-stub"
            />
          </button>
        </a>
      </div>
      <div
        className="footer__links"
      >
        <a
          aria-current="page"
          className="link"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          <img
            alt=""
            className="footer__logo"
            src="test-file-stub"
          />
        </a>
        <div
          className="footer__social-wrapper"
        >
          <a
            className="footer__social"
            href="https://ua.linkedin.com/company/blackbird-lab"
            target="_blank"
          >
            in
          </a>
          <div
            className="footer__bar"
          />
          <a
            className="footer__social"
            href="https://jobs.dou.ua/companies/blackbird-lab/"
            target="_blank"
          >
            DOU
          </a>
        </div>
      </div>
    </div>
    <div
      className="footer__functions"
    >
      <ul
        className="footer__lists"
      >
        <span
          className="footer__small-title"
        >
          Find a Service
        </span>
        <a
          href="/what-we-do/web-development/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            Web development
          </li>
        </a>
        <a>
          <li
            className="footer__link"
          >
            Quality Control
          </li>
        </a>
        <a>
          <li
            className="footer__link"
          >
            Research
          </li>
        </a>
        <a>
          <li
            className="footer__link"
          >
            Analytics
          </li>
        </a>
        <a
          href="/what-we-do/cloud-services/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            Cloud Services
          </li>
        </a>
        <a
          href="/what-we-do/mobile-development/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            Mobile development
          </li>
        </a>
      </ul>
      <ul
        className="footer__lists"
      >
        <span
          className="footer__small-title"
        >
          Pick an Expertise
        </span>
        <a
          href="/what-we-do/retail-and-e-commerce/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            Retail and E-commerce
          </li>
        </a>
        <a>
          <li
            className="footer__link"
          >
            Construction
          </li>
        </a>
        <a>
          <li
            className="footer__link"
          >
            Finance and ERP
          </li>
        </a>
        <a>
          <li
            className="footer__link"
          >
            Media
          </li>
        </a>
        <a
          href="/what-we-do/ar-&-vr/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            AR & VR
          </li>
        </a>
      </ul>
      <ul
        className="footer__lists"
      >
        <span
          className="footer__small-title"
        >
          Company
        </span>
        <a
          href="/about-us/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            About Us
          </li>
        </a>
        <a
          href="/blog/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            Blog and Resources
          </li>
        </a>
        <a>
          <li
            className="footer__link"
          >
            Careers
            <span
              className="footer__label"
            >
              Hiring
            </span>
          </li>
        </a>
        <a
          href="/contact-us/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <li
            className="footer__link"
          >
            Contact Blackbird
          </li>
        </a>
      </ul>
    </div>
  </div>
  <div
    className="small-footer"
    data-testid="secondary-footer"
  >
    <div
      className="small-footer__wrapper"
    >
      <p
        className="small-footer__rights"
      >
        © Blackbird Lab. 2017-2020. All rights reserved.
      </p>
      <div
        className="small-footer__links"
      >
        <a
          className="link"
          href="/acknowledgement-and-acceptance"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            className="small-footer__privacy"
          >
            Privacy Policy
          </span>
        </a>
        <a
          className="small-footer__link"
          href="https://ua.linkedin.com/company/blackbird-lab"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            />
          </svg>
        </a>
        <a
          className="small-footer__link"
          href="https://jobs.dou.ua/companies/blackbird-lab/"
          target="_blank"
        >
          DOU
        </a>
      </div>
    </div>
  </div>
</footer>
`;
